client-sdk/go/callformat/callformat.go

```diff
@@ -82,20 +81,23 @@ func DecodeResult(result *types.CallResult, meta interface{}) (*types.CallResult
 		// In case of plain-text data format, we simply pass on the result unchanged.
 		return result, nil
 	case *metaEncryptedX25519DeoxysII:
-		// Make sure the result makes sense in this context.
-		switch {
-		case result.IsUnknown():
-		case result.IsSuccess():
-			// Unexpected as a successful result shouldn't be plain.
-			return nil, fmt.Errorf("callformat: unexpected result: %s", base64.StdEncoding.EncodeToString(result.Ok))
-		default:
-			// Submission could fail before call format processing so the result would be plain.
+		// Submission could fail before call format processing so the result would be plain.
+		if result.Failed != nil {
 			return nil, result.Failed
 		}
 
 		var envelope types.ResultEnvelopeX25519DeoxysII
-		if err := cbor.Unmarshal(result.Unknown, &envelope); err != nil {
-			return nil, fmt.Errorf("callformat: malformed result envelope: %w", err)
+
+		if result.IsUnknown() {
```

Comment:
Looks like these two would make more sense in a switch as only one of them should be true. Actually the failed one could also be handled here?

Reply:
Ah, fair enough. `switch` it is.